test/integration/protocol_integration_test.cc

```diff
@@ -122,6 +122,57 @@ TEST_P(DownstreamProtocolIntegrationTest, RouterRedirect) {
             response->headers().get(Http::Headers::get().Location)[0]->value().getStringView());
 }
 
+// Add a filter that overrides the cached route/cluster selection with a DelegatingRoute
+TEST_P(ProtocolIntegrationTest, RouterClusterFromDelegatingRoute) {
+  config_helper_.addFilter(R"EOF(
+  name: set-route-filter
+  )EOF");
+
+  // Fake upstream clusters: integration, cluster_0, cluster_override
+  setUpstreamCount(3);
+  config_helper_.addConfigModifier([](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+    auto* co_cluster = bootstrap.mutable_static_resources()->add_clusters();
+    co_cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
+    co_cluster->set_name("cluster_override");
+    co_cluster->set_type(envoy::config::cluster::v3::Cluster::ORIGINAL_DST);
+    // Necessary for ORIGINAL_DST clusters
+    co_cluster->clear_load_assignment();
+    co_cluster->set_lb_policy(envoy::config::cluster::v3::Cluster::CLUSTER_PROVIDED);
+    ConfigHelper::setHttp2(*co_cluster);
+  });
```

Comment:
Is using original dst cluster necessary here? Seems like it complicates the test

Reply:
It's not, but I was hoping to use it since the first use case of this setRoute / DelegatingRoute functionality will be for the Lyft context propagation filter, which will re-route requests to an ORIGINAL_DST cluster